rosbag2_transport/test/rosbag2_transport/memory_management.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
```

Comment:
what is the difference to the memory management in the test folder?

Reply:
There is none, this is the question I posted under "To discuss": We don't have a way to avoid this duplication just now. Ideas are to
a) create test libraries within each package that can be used by dependant packages (i.e. rosbag2_tests would use memory_management from rosbag2_transport)
b) create a `test_commons` library which all libraries can use as a (pure) test dependency. The `rosbag2_tests` package cannot be this package as it contains our end-to-end tests and must therefore itself depend on `ros2bag`.

Therefore, for now, we just duplicated the code. Once we decided on the outcome, we can just delete all duplicate files.